openupgrade_scripts/scripts/project/15.0.1.2/post-migration.py

```diff
@@ -0,0 +1,108 @@
+from openupgradelib import openupgrade
+
+
+def _convert_field_m2o_to_m2m(env):
+    # Convert m2o to m2m in 'project.task'
+    openupgrade.m2o_to_x2m(
+        env.cr,
+        env["project.task"],
+        "project_task",
+        "user_ids",
+        openupgrade.get_legacy_name("user_id"),
+    )
+
+
+def _create_missing_project_task_stage_personal(env):
+    tasks = env["project.task"].with_context(active_test=False).search([])
+    tasks._populate_missing_personal_stages()
+    # update for 'done' and 'cancelled' tasks
+    openupgrade.logged_query(
```

Comment:
This is not correct. Personal stages are not the same as regular ones. They are stages with specified owner. We don't have to populate anything here, as there are no personal stages yet.

Reply:
No, the way it works now is different. Each time a task is created, the method `_populate_missing_personal_stages()` is called:
![Selection_503](https://user-images.githubusercontent.com/25005517/182344961-211a9b69-dd26-4e4c-9b48-27d370b80d8f.png)
Thus we must run this method for the already created tasks to work as they were created in v15. And this last query just ensures that the stage assigned to the done and cancelled tasks receive the corresponding stage instead of the one the method assigns.

Reply:
But I insist: these stages are not created by default as such table is not populated.

Reply:
As I said above, the table is not empty. They are the assignees of the task. And all the stages are created (when calling the _populate_missing_personal_stages):
![Selection_505](https://user-images.githubusercontent.com/25005517/182359240-0a0108d7-9cab-4a44-8d9a-0a0ad5118740.png)
.